priv/apps/itest/test/itest/fee_claiming_test.exs

```diff
@@ -0,0 +1,119 @@
+defmodule FeeClaimingTests do
+  use Cabbage.Feature, async: false, file: "fee_claiming.feature"
+
+  require Logger
+
+  alias Itest.Account
+  alias Itest.Client
+  alias Itest.Transactions.Currency
+
+  @fee_claimer_address "0x3b9f4c1dd26e0be593373b1d36cee2008cbeb837"
+  @expected_fee_rule %{"amount" => 1, "currency" => "0x0000000000000000000000000000000000000000"}
+
+  setup do
+    [{alice_address, alice_pkey}, {bob_address, bob_pkey}] = Account.take_accounts(2)
+
+    initial_balance =
+      @fee_claimer_address
+      |> Client.get_balance()
+      |> fix_balance_response()
+
+    assert_equal(@expected_fee_rule["currency"], initial_balance["currency"], "fee expected currency")
+
+    %{
+      "Alice" => %{
+        address: alice_address,
+        pkey: alice_pkey
+      },
+      "Bob" => %{
+        address: bob_address,
+        pkey: bob_pkey
+      },
+      "FeeClaimer" => %{address: @fee_claimer_address},
+      fees_initial_balance: initial_balance,
+      gas: 0
+    }
+  end
+
+  defwhen ~r/^"(?<entity>[^"]+)" deposits "(?<amount>[^"]+)" ETH to the root chain$/,
+          %{entity: entity, amount: amount},
+          state do
+    entity_address = get_entity(state, entity).address
+
+    {:ok, receipt_hash} =
+      amount
+      |> Currency.to_wei()
+      |> Client.deposit(entity_address, Itest.Account.vault(Currency.ether()))
+
+    gas_used = Client.get_gas_used(receipt_hash)
+
+    {_, new_state} =
+      Map.get_and_update!(state, :gas, fn current_gas ->
+        {current_gas, current_gas + gas_used}
+      end)
+
+    {:ok, new_state}
+  end
+
+  defwhen ~r/^"(?<sender>[^"]+)" sends "(?<receiver>[^"]+)" "(?<amount>[^"]+)" ETH on the child chain$/,
+          %{sender: sender, receiver: receiver, amount: amount},
+          state do
+    sender = get_entity(state, sender)
+    receiver = get_entity(state, receiver)
+
+    {:ok, [sign_hash, typed_data, _txbytes]} =
+      Client.create_transaction(
+        Currency.to_wei(amount),
+        sender.address,
+        receiver.address
+      )
+
+    _ = Client.submit_transaction(typed_data, sign_hash, [sender.pkey])
+
+    {:ok, state}
+  end
+
+  defthen ~r/^"(?<entity>[^"]+)" should have "(?<amount>[^"]+)" ETH on the child chain$/,
+          %{entity: entity, amount: amount},
+          state do
+    entity_address = get_entity(state, entity).address
+    expecting_amount = Currency.to_wei(amount)
+
+    balance =
+      entity_address
+      |> Client.get_balance(expecting_amount)
+      |> fix_balance_response()
```

Comment:
I would suggest, for the sake of consistency, that we handle this malformed response the same way as in other tests, even if not optimal
```
child_chain_balance =
      case amount do
        "0" ->
          assert Client.get_balance(address, Currency.to_wei(amount)) == []
          0

        _ ->
          %{"amount" => network_amount} = Client.get_balance(address, Currency.to_wei(amount))
          assert network_amount == Currency.to_wei(amount)
          network_amount
      end

```

or let's change it everywhere else the way you did it here.

Reply:
IMHO we can fix it for all #1293 - and what you posted isn't this exact case (there are 3 clauses of `fix_balance_response()`

Reply:
I guess the response depends on whether you provide the amount you're looking for Client.get_balance/2 or `Client.get_balance/1`